executor/join.go

```diff
@@ -532,20 +558,23 @@ func (e *HashJoinExec) buildHashTableForList() error {
 		keyBuf  = make([]byte, 0, 64)
 		valBuf  = make([]byte, 8)
 	)
-	for i := 0; i < e.innerResult.NumChunks(); i++ {
-		chk := e.innerResult.GetChunk(i)
-		for j := 0; j < chk.NumRows(); j++ {
+
+	numChks := 0
+	for chk := range chkCh {
+		numRows := chk.NumRows()
+		for j := 0; j < numRows; j++ {
```

Comment:
j < chk.NumRows()

Reply:
```go
func BenchmarkFor1(b *testing.B) {
	chk := getChks()
	for i := 0; i < b.N; i++ {
		for j := 0; j < chk.NumRows(); j++ {
			_ = j
		}
	}
}
func BenchmarkFor2(b *testing.B) {
	chk := getChks()
	for i := 0; i < b.N; i++ {
		numRows := chk.NumRows()
		for j := 0; j < numRows; j++ {
			_ = j
		}
	}
}
```
chk.NumRows is 1024.
```
BenchmarkFor1    5000000              1115 ns/op
BenchmarkFor1    5000000              1119 ns/op
BenchmarkFor1    5000000              1106 ns/op
BenchmarkFor2   10000000               557 ns/op
BenchmarkFor2   10000000               577 ns/op
BenchmarkFor2   10000000               587 ns/op
```
so, I think the original is better.